scripts/validate.ts

```diff
@@ -0,0 +1,21 @@
+import Ajv from "ajv";
+import addFormats from "ajv-formats";
+import assert from "node:assert/strict";
+
+import * as schema from "../schemas/data.schema.json" with { type: "json" };
+
+export function validate(data: any) {
+  const ajv = new Ajv({ allErrors: true, allowUnionTypes: true });
```

Comment:
I consolidated compiling the schema here, so that we can have one place with settings (and to check that we don't do something foolish like validate nothing at all).